### ISSUE TITLE
Option to only include dependencies; exclude devDependencies

### ISSUE THREAD

**Issue by 64bit** [TobyAndToby/generate-license-file#163]
Hello, This is very useful project!

I wonder if including any other dependency in generated license file should include `devDependencies` or other types of dependencies listed in https://docs.npmjs.com/cli/v9/configuring-npm/package-json. 

My understanding is that  final build output distributed would not include `devDependencies` ? 

I was not able to disable licenses of devDependencies in generated file using current cli. It would be nice to have cli option to not include devDependencies. Also wondering if there are workaround in existing cli to do so?

Thank you

**Comment by tobysmith568** [comment]
Hey @64bit ,

Generate-License-File should already be filtering out dev dependencies for you.

Are you finding that your output does contain dev dependencies?
If so, would you mind please sharing a `package.json` and `package-lock.json` repro with us?

Thanks!

**Comment by 64bit** [comment]
That's wonderful! 

I have attached zip file containing both json files (Github doesn't  let me upload json file directly). 

Here's my observation: In generated license txt file I see Creative Commons for `caniuse-lite` package then running `npm ls caniuse-lite` shows it comes from `autoprefixer` (and root?)  which is a `devDependencies` and hence the reported Issue:

```
gui@0.0.0 /home/himanshu/license-generator
└─┬ autoprefixer@10.4.14
  ├─┬ browserslist@4.21.5
  │ └── caniuse-lite@1.0.30001473 deduped
  └── caniuse-lite@1.0.30001473
```


[package-files.zip](https://github.com/TobyAndToby/generate-license-file/files/11730057/package-files.zip)

**Comment by tobysmith568** [comment]
Ah okay, so I think I've gotten to the bottom of this.

In your package-lock.json on line 1311 you can see that `autoprefixer@10.4.14` lists `caniuse-lite@1.0.30001464` as a _normal_ dependency, not a devDependency. However, on line 30 you can see that `autoprefixer@10.4.14` is listed as a _devDependency_ of your project, meaning in theory that `caniuse-lite@1.0.30001464` is therefore in turn a devDependency from the perspective of your project.

The problem (and what `npm ls` seems to have not shown you) is that on line 1606 you can see that `autoprefixer@10.0.2` (that gets deduped to `autoprefixer@10.4.14`) is a _peerDependency_ of `daisyui@2.51.5`. If you follow that up the dependency tree to line 12 then you can see `daisyui` is listed as one of your normal dependencies. Through this route (`caniuse-lite` <- `autoprefixer` <- `daisyui`) `caniuse-lite` is not a "pure" devDependency.

For me this raises two questions:
1) Should generate-license-file show the licenses for packages that list those you depend on as peerDependencies?
2) Should `daisyui` list `autoprefixer` as a peerDependency?

For 1) I'm not 100% sure but I'm leaning towards "yes". If your project has a _normal_ dependency on `package-a` and `package-b` lists `package-a` as a peerDependency then that implies "while `package-a` does not directly require/import from `package-b`, it will discover it dynamically and run its code". To me that means the code from `package-b` is run in your project and should be listed.

For 2) I'm also not 100% sure but I'm learning towards "no". I'm not familiar with `daisyui` but from what I can see:
- You use `daisyui` version 2.51.5
- Back in version 2.51.5 [they listed `autoprefixer` as a peerDependency](https://github.com/saadeghi/daisyui/blob/b1661b51fc16ccb13c6b97ab02007cac4f845cf1/package.json#L115)
- However [they only ever required/imported it as a devDependency](https://github.com/saadeghi/daisyui/blob/b1661b51fc16ccb13c6b97ab02007cac4f845cf1/src/components/postcss.config.js#L8)
- In their current version (3.1.0) [they've removed `autoprefixer` as a peerDependency](https://github.com/saadeghi/daisyui/blob/master/package.json#L112) and it's [just a DevDependency](https://github.com/saadeghi/daisyui/blob/master/package.json#L95)

In conclusion, I believe that in reality `caniuse-lite` is a devDependency of your project but generate-license-file is treating it as a normal dependency because `daisyui` used to incorrectly list `autoprefixer` as a peerDependency. If you were to update `daisyui` to 3.1.0 then I believe generate-license-file would stop showing `caniuse-lite`.

**Comment by 64bit** [comment]
Hey @tobysmith568 , Truly appreciate your humongous effort to see what's going on and explaining in detail!

Your reasoning makes sense to me, the responsibility should be on packages to list dependency accurately.  Fortunately, `caniuse-lite` doesn't put NC , ND restrictions so its not a blocker and I can update to `daisyui` 3.1.0 when time comes. 

I'm closing this since generate-license-file is working as expected. 

Thank you!
